abigen/src/lib.rs

```diff
@@ -138,10 +137,13 @@ fn fixed_data_size(input: &ParamType) -> Option<usize> {
         ParamType::FixedArray(ref sub_type, count) => {
             Some(count * fixed_data_size(sub_type).expect("not dynamic, will always be Some(_)"))
         }
-        ParamType::Tuple(_) => {
-            unimplemented!(
-                "Tuples are not supported. https://github.com/openethereum/ethabi/issues/175"
-            )
+        ParamType::Tuple(ref types) => {
+            let sizes = types.iter().map(fixed_data_size);
+            if sizes.clone().any(|x| x.is_none()) {
+                None
+            } else {
+                Some(sizes.map(Option::unwrap).sum())
+            }
```

Comment:
I prefer that we do 2 iteration instead of doing a `clone` here
```suggestion
            if types.iter().any(ParamType::is_dynamic) {
                return None;
            }

            Some(types.iter().map(fixed_data_size).map(Option::unwrap).sum())
```